portal_tables/sync_tools.py

```diff
@@ -5,32 +5,49 @@
 """
 
 import pandas as pd
+import re
 import utils
 
 
 def add_missing_info(tools: pd.DataFrame, grants: pd.DataFrame) -> pd.DataFrame:
     """Add missing information into table before syncing."""
+    url_pattern = re.compile(".*(synapse\.org).*")
     tools["link"] = "[Link](" + tools["ToolHomepage"] + ")"
     tools["portalDisplay"] = "true"
     tools["themes"] = ""
     tools["consortium"] = ""
+    tools["synapseLink"] = ""
     for _, row in tools.iterrows():
         themes = set()
         consortium = set()
-        for g in row["ToolGrantNumber"].split(","):
+        for g in row["GrantViewKey"].split(","):
             if g not in ["", "Affiliated/Non-Grant Associated"]:
                 themes.update(grants[grants.grantNumber == g]["theme"].values[0])
                 consortium.update(
                     grants[grants.grantNumber == g]["consortium"].values[0]
                 )
         tools.at[_, "themes"] = list(themes)
         tools.at[_, "consortium"] = list(consortium)
+        
+        synapse_links = []
+        for s in [row["ToolDownloadUrl"], row["ToolLinkUrl"], row["ToolHomepage"]]:
+            s_match = re.match(url_pattern, s)
+            if s_match:
+                print(s)
+                synapse_links.append("".join(["[Link](", s , ")"]))
+        tools.at[_, "synapseLink"] = ", ".join(list(set(synapse_links)))
```

Comment:
No need to type cast back to a list, since sets are also iterable :-)

```suggestion
        tools.at[_, "synapseLink"] = ", ".join(set(synapse_links))
```

Reply:
Makes sense!